### PR TITLE
Datepicker auto ok

### LINKED ISSUE #342: [DatePicker] Select a date by clicking a number?
Is there a possibility to select a date by clicking a number, and not `OK` button? Like in Material-UI `autoOk={true}` property. So no need `OK` button for this behavior.

And how to close DP dialog by clicking outside of it? Now it closes only if you click `Close` button inside DP dialog.

### PR DESCRIPTION
**Pull request by felipeleusin**
Provides an autoOk to the DatePicker that auto selects a date upon clicking on a day.

Closes #342 

I based it off #366 so that should be merged first.

### CODE REVIEW

**Comment by javivelasco** [comment]
May you refresh the branch with the previous changes for the date format. That way we can see the diff with more precision! :)

File: components/dialog/Dialog.jsx [hunk 1]
```diff
@@ -42,8 +42,8 @@ Dialog.propTypes = {
   className: React.PropTypes.string,
   onOverlayClick: React.PropTypes.func,
   onOverlayMouseDown: React.PropTypes.func,
-  onOverlayMouseUp: React.PropTypes.func,
   onOverlayMouseMove: React.PropTypes.func,
+  onOverlayMouseUp: React.PropTypes.func,
```
Comment (felipeleusin) [inline components/dialog/Dialog.jsx]: This was just to fix a lint error

**Comment by felipeleusin** [comment]
Done

**Comment by javivelasco** [comment]
Great!

**Pull request merged**